ansible/roles/test/files/ptftests/copp_tests.py

```diff
@@ -347,6 +352,76 @@ def contruct_packet(self, port_number):
 
         return packet
 
+class DHCPIP2METest(NoPolicyTest):
+    def __init__(self):
+        NoPolicyTest.__init__(self)
+
+    def runTest(self):
+        self.log("DHCPIP2METest")
+        self.run_suite()
+
+    def contruct_packet(self, port_number):
+        src_mac = self.my_mac[port_number]
+        dst_mac = self.peer_mac[port_number]
+        dst_ip = self.peerip
+
+        packet = testutils.simple_udp_packet(
+            pktlen=100,
+            eth_src=src_mac,
+            eth_dst=dst_mac,
+            dl_vlan_enable=False,
+            vlan_vid=0,
+            vlan_pcp=0,
+            dl_vlan_cfi=0,
+            ip_src='0.0.0.0',
+            ip_dst=dst_ip,
+            ip_tos=0,
+            ip_ttl=64,
+            udp_sport=68,
+            udp_dport=67,
+            ip_ihl=None,
+            ip_options=False,
+            with_udp_chksum=True
+        )
+
+        return packet
+
+class DHCPOTHERIPTest(PolicyTest):
+    def __init__(self):
+        PolicyTest.__init__(self)
+        # Police rate is expected to be 0
+        self.PPS_LIMIT_MIN = 0
+        self.PPS_LIMIT_MAX = 0
+
+    def runTest(self):
+        self.log("DHCPOTHERIPTest")
+        self.run_suite()
+
+    def contruct_packet(self, port_number):
+        src_mac = self.my_mac[port_number]
+        dst_mac = self.peer_mac[port_number]
+
+        packet = testutils.simple_udp_packet(
+            pktlen=100,
+            eth_dst=dst_mac,
+            eth_src=src_mac,
+            dl_vlan_enable=False,
+            vlan_vid=0,
+            vlan_pcp=0,
+            dl_vlan_cfi=0,
+            ip_src='0.0.0.0',
+            ip_dst=socket.inet_ntoa(struct.pack('>I', random.randint(1, 0xffffffff))),
```

Comment:
is there possibility random ip can me ip2me ?

Reply:
This is meant to be a negative testcase for not IP2ME/Broadcast. Is there a better alternative test besides random IP?

Reply:
what i meant is it possible that above random function can generate ip2me ? if so you can use peer ip of  self.myip since it is /31

Reply:
I see. Updated randip to exclude ip2me